src/components/TrackingConsent.vue

```diff
@@ -0,0 +1,467 @@
+<template>
+    <div class="tracking-consent"
+        :class="[ theme ]"
+        v-if="uiRequired && uiAllowed"
+    >
+        <div class="content nq-shadow">
+            {{ text.main }}
+            <div class="button-group">
+                <button
+                    class="nq-button-pill light-blue"
+                    @click="allowUsageData"
+                >{{ text.yes }}</button>
+                <button
+                    class="nq-button-s"
+                    @click="denyConsent"
+                    :class="{ inverse: theme === constructor.Themes.DARK }"
+                >{{ text.no }}</button>
+                <button
+                    class="nq-button-s"
+                    @click="allowBrowserData"
+                    :class="{ inverse: theme === constructor.Themes.DARK }"
+                >{{ text.browserOnly }}</button>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+import I18nMixin from '../i18n/I18nMixin';
+
+interface Consent {
+    allowsBrowserData?: boolean;
+    allowsUsageData?: boolean;
+}
+/**
+ * TrackinConsent Component - used to setup Matomo tracking in a vue-based project
+ */
+@Component
+class TrackingConsent extends Vue {
+    private static _instances: Set<TrackingConsent> = new Set();
+
+    @Prop({
+        type: Object,
+        default:  () => ({
+            get main() {
+                return I18nMixin.$t(
+                    'TrackingConsent',
+                    'Help Nimiq improve by sharing anonymized usage data. Thank you! ❤️',
+                );
+            },
+            get yes() { return I18nMixin.$t('TrackingConsent', 'Yes'); },
+            get no() { return I18nMixin.$t('TrackingConsent', 'No'); },
+            browserOnly: 'Browser-info only',
+        }),
+        validator: (text) => (
+            text && typeof text === 'object' &&
+            text.no && typeof text.no === 'string' && text.no.length &&
+            text.yes && typeof text.yes === 'string' && text.yes.length &&
+            text.main && typeof text.main === 'string' && text.main.length &&
+            text.browserOnly && typeof text.browserOnly === 'string' && text.browserOnly.length
+        ),
+    })
+    public text: {
+        main: string,
+        yes: string,
+        no: string,
+        browserOnly: string,
+    };
+
+    @Prop({
+        type: String,
+        default: 'light',
+        validator: (theme) => Object.values(TrackingConsent.Themes).includes(theme),
+    })
+    public theme: string;
+
+    /** API reference: https://developer.matomo.org/guides/tracking-javascript#configuration-of-the-tracker-object */
+    @Prop({
+        type: Object,
+        default: () => ({
+            setSiteId: 1,
```

Comment:
I would not set a default siteId. This is a _bad default_ and will lead to tracking on sites that don't change this. This prop should be `required` and the sideId should be a required setting.